### PR TITLE
feat: adds github issue command

### LINKED ISSUE #28: Feature Request - Issues command to list github issues
### Feature type

Utility or complex functionality

### Description

Similar to the PR command, but for github issues instead

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

straks

### PR DESCRIPTION
**Pull request by benw202**
## Description

Fixes #28 

Adds the github issue command and consolidates PR and Issue into one command group `/github`

## Test Results

image

## Discord Username

benw8484

### CODE REVIEW

File: src/commands/utils/github/functions/handleGithubIssue.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { ChatInputCommandInteraction, Colors } from 'discord.js';
+import { request } from '@octokit/request';
+import { makeEmbed } from '../../../../lib';
+
+const syntaxHelp = '\nSyntax:\nA32NX repo: `/github issue <id>`\nAny FBW repo: `/github issue <repo> <id>`';
```
Comment (pdellaert) [inline src/commands/utils/github/functions/handleGithubIssue.ts]:
```suggestion
const syntaxHelp = '\nSyntax:\nAircraft repo: `/github issue <id>`\nAny FBW repo: `/github issue <repo> <id>`';
```

File: src/commands/utils/github/functions/githubPullRequest.ts [hunk 2]
```diff
@@ -1,29 +1,8 @@
...
-    {
-        name: 'repo',
-        description: 'Please provide the repo name.',
-        type: ApplicationCommandOptionType.String,
-        max_length: 100,
-        required: false,
-    },
-    ],
-});
+const syntaxHelp = '\nSyntax:\nA32NX repo: `/github pr <id>`\nAny FBW repo: `/github pr <repo> <id>`';
```
Comment (pdellaert) [inline src/commands/utils/github/functions/githubPullRequest.ts]:
```suggestion
const syntaxHelp = '\nSyntax:\nAircraft repo: `/github pr <id>`\nAny FBW repo: `/github pr <repo> <id>`';
```

File: src/commands/utils/github/functions/handleGithubIssue.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    if (!issueNumber) return interaction.reply({ embeds: [noQueryEmbed] });
+
+    const cleanedIssueNumber = issueNumber.replace('#', '');
+
+    if (repoName) {
+        try {
+            const response = await request('GET /repos/flybywiresim/{repo}/issues/{pull_number}', {
+                repo: repoName,
+                pull_number: cleanedIssueNumber,
```
Comment (pdellaert) [inline src/commands/utils/github/functions/handleGithubIssue.ts]:
```suggestion
            const response = await request('GET /repos/flybywiresim/{repo}/issues/{issue_number}', {
                repo: repoName,
                issue_number: cleanedIssueNumber,
```
Just a bit cleaner in context

File: src/commands/utils/github/functions/handleGithubIssue.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            });
+            return interaction.reply(response.data.html_url);
+        } catch {
+            return interaction.reply({ embeds: [invalidEmbed] });
+        }
+    } else {
+        try {
+            const response = await request('GET /repos/flybywiresim/a32nx/issues/{pull_number}', { pull_number: cleanedIssueNumber });
```
Comment (pdellaert) [inline src/commands/utils/github/functions/handleGithubIssue.ts]:
```suggestion
            const response = await request('GET /repos/flybywiresim/a32nx/issues/{issue_number}', { issue_number: cleanedIssueNumber });
```

**Review by pdellaert — CHANGES_REQUESTED** [review]
Just a proposal for some minor changes. Great stuff :)

File: src/commands/utils/github/functions/handleGithubIssue.ts (hunk 3 above)
Comment (benw202) [reply src/commands/utils/github/functions/handleGithubIssue.ts]: Totally missed these ones in my dumb copy paste! Thanks

File: src/commands/utils/github/functions/handleGithubIssue.ts (hunk 1 above)
Comment (benw202) [reply src/commands/utils/github/functions/handleGithubIssue.ts]: Ahh yes! Thanks

**Review by pdellaert — APPROVED** [review]

**Pull request merged**